catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java

```diff
@@ -0,0 +1,185 @@
+ **/
+
+package org.codice.ddf.registry.federationadmin.service.impl;
+
+import static org.codice.ddf.registry.schemabindings.EbrimConstants.RIM_FACTORY;
+
+import java.io.IOException;
+import java.security.PrivilegedActionException;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.UUID;
+
+import javax.xml.datatype.DatatypeConstants;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.configuration.SystemBaseUrl;
+import org.codice.ddf.configuration.SystemInfo;
+import org.codice.ddf.parser.ParserException;
+import org.codice.ddf.registry.common.RegistryConstants;
+import org.codice.ddf.registry.common.metacard.RegistryObjectMetacardType;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminException;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminService;
+import org.codice.ddf.registry.schemabindings.helper.InternationalStringTypeHelper;
+import org.codice.ddf.registry.schemabindings.helper.MetacardMarshaller;
+import org.codice.ddf.registry.schemabindings.helper.SlotTypeHelper;
+import org.codice.ddf.security.common.Security;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ExtrinsicObjectType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryPackageType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.VersionInfoType;
+
+/**
+ * Creates a registry identity node when DDF Registry is first instantiated.
+ * If a previous registry node is not already found, then a registry identity metacard and its
+ * attributes are created. FederationAdminService adds the newly created identity metacard as a registry entry.
+ */
+public class IdentityNodeInitialization {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IdentityNodeInitialization.class);
+
+    private FederationAdminService federationAdminService;
+
+    private InputTransformer registryTransformer;
+
+    private MetacardMarshaller metacardMarshaller;
+
+    private SlotTypeHelper slotTypeHelper = new SlotTypeHelper();
+
+    private InternationalStringTypeHelper internationalStringTypeHelper =
+            new InternationalStringTypeHelper();
+
+    public void init() throws FederationAdminException {
+        try {
+            Security.runAsAdminWithException(() -> {
+                if (!federationAdminService.getLocalRegistryIdentityMetacard()
+                        .isPresent()) {
+                    createIdentityNode();
+                }
+                return null;
+            });
+        } catch (PrivilegedActionException e) {
+            String message = "There was an error bringing up the Federation Admin Service.";
```

Comment:
This message needs to be updated. This isn't the Federation Admin Service anymore